### DISCUSSION TITLE
Jest test help

### DISCUSSION THREAD

**Discussion by owangen** [Koenkk/zigbee2mqtt#8743] [General]
I have started helping out developing converts for various devices. I think it's fun and rewarding helping out. I have a development machine with Windows with vscode installed. It works fine, but I have problem running the jest test. When I run npm test in the zigbee-herdsman-converts it triggers "ZHC_TEST=true jest test", but complains about "ZHC_TEST' is not recognized as an internal or external command, operable program or batch file." 

Anyone developing with Windows and vscode and have jest test are working? I would be happy if I could get some help here.

**Comment by Koenkk** [comment]
Can you try with the git bash on Windows?

**Reply by owangen** [reply]
Thanks for the help, but I'm getting the same result when using the git bash from vscode on Windows. I'm new to jest overall, but I'm using the vscode-jest extension on vscode, so I thought it should be smooth sailing.

**Reply by pgfeller** [reply]
I've the same problem (on Windows OS) - I'm able to run the tests if I modify the npm scripts:

```json
  "scripts": {
    "lint": "eslint .",
    "test": "set ZHC_TEST=true && jest test",
    "test-watch": "set ZHC_TEST=true && jest test --watch"
  },
```
nothing to be checked in - but a temporary solution for the moment.

**Comment by owangen** [comment]
No one uses vscode on Windows here? 
What are your prefered IDE and OS, while developing for zigbee2mqtt? I'm adaptable :-)

**Reply by Koenkk** [reply]
Linux should definitely work (I develop on Mac)

**Reply by owangen** [reply]
If anyone else have this question, I solved it using [cross-env](https://www.npmjs.com/package/cross-env) ref https://github.com/jest-community/vscode-jest/discussions/765#discussioncomment-1661112

**Comment by xamadevp** [comment]
I'm using Ubuntu with WSL 2 and [installed node.js](https://learn.microsoft.com/en-us/windows/dev-environment/javascript/nodejs-on-wsl) there. Running `npm test` works. My first setup was node.js on Windows but it didn't work properly in WSL. Hope this helps.

**Comment by owangen** [comment]
Thanks. Yes, I ended up using Ubuntu in a virtualbox. This works fine.
Could not get WSL to work together with Zigbee adapter.


man. 13. mar. 2023, 15:07 skrev xamadevp ***@***.***>:
